src/dvc_data/index/index.py

```diff
@@ -119,7 +119,7 @@ def _try_load(
             continue
 
         try:
-            return Tree.load(odb, hash_info)
+            return Tree.load(odb, hash_info, hash_name="md5")
```

Comment:
This is a bit of a strong assumption, but this will not bite us until we get into supporting alternative hash algos, which will be a whole separate journey.

Reply:
Though if the way we serialize trees wouldn't just dump hash_info and meta together - we wouldn't even have to pass or have this flag. Probably a topic for a new `Tree` object a bit later on.